### PR TITLE
BFD Enhancement

### PR DESCRIPTION
**Pull request by SumitAgarwal123**
This document describes design for a few enhancements to BFD in BGP docker.

Proposes enhancement to Packet Tx.
A few display and debug enhancements.

### CODE REVIEW

File: bfd/BFD_Enhancement_HLD.md [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+
+A new show command is added as below:
+
+**show bfd peers brief**
+Show all the BFD peer in brief, sample output as below:
+This command is available in FRR vtysh shell.
+
+``` 
+Session count: 1
+OurAddr                  NeighAddr                        State
```
Comment (sayedsaquibbrcm) [inline bfd/BFD_Enhancement_HLD.md]: Pls change OurAddr to LocalAddr maybe.
Can you include a column for BFD session Id as well

Comment (SumitAgarwal123) [reply bfd/BFD_Enhancement_HLD.md]: Will update

File: bfd/BFD_Enhancement_HLD.md [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+
+Supporting ECMP is challenging in BFD due to time it may take for control plane to know that the active nexthop went down. In SONiC this information has to traverse all the way down to kernel after traversing all the DBs this may take considerable time.                                                                         
+
+In current SONiC implementation BFD relies on kernel network stack to switch the BFD packet to next available nexthop when a active nexthop goes down. BFD timers in this case is directly proportional to the time it takes for kernel to get the nexthop down event. Faster the kernel learns active nexthop is down the more aggressive BFD timers can be. In this case it is suggested to configure BFD timer values to have a timeout of atleast 600 msec.                                                                                                                                            
+
+## 3.2 CLI
```
[the remark this answers is not in the document]
Comment (SumitAgarwal123) [reply bfd/BFD_Enhancement_HLD.md]: A new change is added in section 8, that should answer your question

**Comment by rck-innovium** [comment]
Today I couldn't find any BFD usage in sonic-swss. Are any SAI BFD APIs planned to be used or is it all managed in control plane?

**Comment by SumitAgarwal123** [comment]
> Today I couldn't find any BFD usage in sonic-swss. Are any SAI BFD APIs planned to be used or is it all managed in control plane?

As for now it is managed in control plane.

File: bfd/BFD_Enhancement_HLD.md [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+"cir":"60000",
+"cbs":"60000",
+"red_action":"drop"
+},
+"OP": "SET"
+}
+```
+
+# 7 Scalability
+ - No of sessions: 64
```
Comment (heidinet2007) [inline bfd/BFD_Enhancement_HLD.md]: Is 64 a hard limit? Do you have any test result for the maximum number of sessions?

Comment (SumitAgarwal123) [reply bfd/BFD_Enhancement_HLD.md]: No 64 it's not a hard limit. We have tested with 64 session but i do not have log to attach.

**Pull request merged**